### PR TITLE
Linked back to query granularity docs

### LINKED ISSUE #12726: Improve documentation around granularities
The possible types of granularities that can be used are only mentioned within the querying section here:
https://druid.apache.org/docs/latest/querying/granularities.html

It's possible to use the same set of granularities as segment granularity during ingestion. We need to clarify this in the docs. Maybe just update the title of this page and link to it from segment granularities too?

### PR DESCRIPTION
**Pull request by hevansDev**
Fixes #12726

### Description

Linked back to query granularity docs from ingestion granularity and storage granularity as query granularity docs are more comprehensive in covering the topic

For tips about how to write a good release note, see [Release notes](https://github.com/apache/druid/blob/master/CONTRIBUTING.md#release-notes).

-->

This PR has:

- [x] been self-reviewed.

### CODE REVIEW

File: docs/design/storage.md [hunk 1]
```diff
@@ -67,7 +67,7 @@ On the Coordinator / Historical side:
 Segments all have a four-part identifier with the following components:
 
 - Datasource name.
-- Time interval (for the time chunk containing the segment; this corresponds to the `segmentGranularity` specified at ingestion time).
+- Time interval (for the time chunk containing the segment; this corresponds to the `segmentGranularity` specified at ingestion time the same as for [query granularity](../querying/granularities.md)).
```
Comment (techdocsmith) [inline docs/design/storage.md:70]:
```suggestion
- Time interval for the time chunk containing the segment; this corresponds to the `segmentGranularity` specified at ingestion time. Uses the same format as [query granularity](../querying/granularities.md).
```
avoid parens. minor

File: docs/ingestion/ingestion-spec.md [hunk 2]
```diff
@@ -324,7 +324,7 @@ An example `granularitySpec` is:
 }
 ```
 
-A `granularitySpec` can have the following components:
+A `granularitySpec` uses the same format as [query granularity](../querying/granularities.md) can have the following components:
```
Comment (techdocsmith) [inline docs/ingestion/ingestion-spec.md:327]:
```suggestion
A `granularitySpec` can have the following components:
```
wrong spot for this :D

File: docs/querying/granularities.md [hunk 3]
```diff
@@ -30,7 +30,7 @@ sidebar_label: "Granularities"
  [SQL documentation](sql-scalar.md#date-and-time-functions).
 :::
 
-Granularity determines how to bucket data across the time dimension, or how to aggregate data by hour, day, minute, etc.
+Granularity determines how to bucket data across the time dimension, or how to aggregate data by hour, day, minute, etc and defines how it is stored.
```
Comment (techdocsmith) [inline docs/querying/granularities.md:33]:
```suggestion
Granularity determines how to bucket data across the time dimension, or how to aggregate data by hour, day, minute, etc and defines how it is stored.
The granularity formats here apply also to `segmentGranularity` and `queryGranularity` in the `granularitySpec` section of the the [ingestion spec](../ingestion/ingestion-spec.md#granularityspec).
```

**Review by techdocsmith — CHANGES_REQUESTED** [review]
some minor suggestions.

File: docs/ingestion/ingestion-spec.md (hunk 2 above)
Comment (hevansDev) [reply docs/ingestion/ingestion-spec.md:327]: Oops! Sorry 😅

Comment (hevansDev) [reply docs/ingestion/ingestion-spec.md]: @techdocsmith where should this line be?

File: docs/ingestion/ingestion-spec.md [hunk 4]
```diff
@@ -301,15 +301,15 @@ An example `metricsSpec` is:
...
-2. Truncating the timestamp, if desired (via `queryGranularity`).
-3. Specifying which time chunks of segments should be created, for batch ingestion (via `intervals`).
-4. Specifying whether ingestion-time [rollup](./rollup.md) should be used or not (via `rollup`).
+1. `segmentGranularity` to partitioning a datasource into [time chunks](../design/storage.md).
+2. `queryGranularity` to optionally truncate the timestamp.
+3. `intervals` to define the time chunks of segments to create for batch ingestion.
+4.  `rollup` to enable ingestion-time [rollup](./rollup.md) or not.
 
 Other than `rollup`, these operations are all based on the [primary timestamp](./schema-model.md#primary-timestamp).
+Use the format in [Query granularities] to specify both `segmentGranualarity` and `queryGranularity`.
```
Comment (techdocsmith) [inline docs/ingestion/ingestion-spec.md:312]:
```suggestion
Use the format from [Query granularities] to specify both `segmentGranualarity` and `queryGranularity`.
```
nit

**Review by techdocsmith — APPROVED** [review]
LGTM

**Pull request merged**